fqlx/queries/remove.js

```diff
@@ -0,0 +1,77 @@
+const { fql, Module } = require('fauna')
+
+/**
+ * Constructs an FQL Query to remove resources managed by this plugin, excluding
+ * those passed as an argument.
+ *
+ * @param module
+ * @param resources An array of definitions that should be excluded from removal.
+ * @returns A Query that deletes functions managed by this plugin. The query returns a Fauna
+ *          Set: Set[{ type: "Function", name: "MyDeletedFunc", result: "deleted" }, ...]
+ */
+const removeExcept  = (module, resources) => {
+  let names = resources.map(f => f.name)
+  return fql`
+  {
+    let mod = ${module}
+    mod.where(
+      f =>
+        !${names}.includes(f.name) &&
+        f.data?.deletion_policy != "retain" &&
+        f.data?.created_by_serverless_plugin == "fauna:v10"
+    ).order(.name).map(f => {
```

Comment:
This `order` call is going to collect up the whole set. So, you could add `toArray()` to the end of this, and it would have no performance impact, and it would make the client code much simpler (you could avoid paginating entirely).

On the other hand, we might fix `order` to do the right thing, but that won't be any time soon. Since you already have the pagination setup, I would leave it as-is, but if pagination ever becomes difficult to work with on the client, this can easily be switched to an array with no downsides.

Reply:
Ah, good to know, ty ty